virtool/users/db.py

```diff
@@ -336,3 +336,27 @@ async def fetch_complete_user(mongo, user_id: str) -> Optional[User]:
         )
 
     return None
+
+
+def lookup_user_by_id(local_field: str, set_as: str = "user") -> list[Dict]:
```

Comment:
Should be named `lookup_nested_user_by_id` Since the lookup results in the addition of a `nestedUser` rather than a full `user` object

Also while technically correct, I would prefer a more specific return type than `list[dict]` there may be something defined as part of pymotor. On a related note this function should have a the summarises what it is for and the inputs and outputs of the function. The docstring should make it clear that this function returns the aggregation step to be put into an aggregation pipeline and not the data itself.